jni/src/common/Foreign/JNI/Unsafe.hs

```diff
@@ -661,12 +682,26 @@ SET_STATIC_FIELD(Long, Int64, jlong)
 SET_STATIC_FIELD(Float, Float, jfloat)
 SET_STATIC_FIELD(Double, Double, jdouble)
 
+-- | wrapper around getMethodID' : raise a verbose exception if no method was found
 getMethodID
   :: JClass -- ^ A class object as returned by 'findClass'
   -> JNI.String -- ^ Field name
   -> MethodSignature -- ^ JNI signature
   -> IO JMethodID
-getMethodID cls methodname (coerce -> sig) = throwIfJNull cls $
+getMethodID cls methodName sig = catch
+  (getMethodID' cls methodName sig)
+  (\(_::JVMException) -> do
```

Comment:
Here we should test whether the exception is really "NoSuchMethodError" and only build our error message if so.